source/common/filter/http/filter_config_discovery_impl.h

```diff
@@ -125,6 +124,8 @@ class FilterConfigSubscription
   const std::string subscription_id_;
   absl::flat_hash_set<DynamicFilterConfigProviderImpl*> filter_config_providers_;
   friend class DynamicFilterConfigProviderImpl;
+
+  std::unique_ptr<Config::Subscription> subscription_;
```

Comment:
Thanks for the quick fix here. As this is a real bug and it's not immediately clear to me what the problem is, can you please do two things:
1) Add some comments here about ordering dependencies.
2) Can you add a unit test that would crash without this change?

Reply:
It's the stats scope. gRPC stream posts some stats on drainage, and it drains on destruction. Will do.

Reply:
It's another race. gRPC stream needs to stay alive on the server while the client closes it, to enable draining and the crash. Trying to find a way to emulate server closure while holding test alive.